### PR TITLE
feat: deprecate node v14

### PR DESCRIPTION
**Pull request by viceice**
## Changes
 - [x] soft node v14 deprecation
 - [x] bump dev container node image

## Context
 preparation for node v18 bekoming lts in two months

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
V33?

**Comment by viceice** [comment]
> V33?

yes, it's next major, we've another PR pending for it

File: package.json [hunk 1]
```diff
@@ -130,7 +130,7 @@
   },
   "engines-next": {
     "description": "Versions other than the below are deprecated and a warning will be logged",
-    "node": "^14.15.0 || >=16.13.0"
+    "node": ">=16.13.0"
```
Comment (HonkingGoose) [inline package.json:133]: You should probably also bump the Node version in the Dockerfile here:

https://github.com/renovatebot/renovate/blob/8ad1947e4a1b334aaf9dc06a87e0c57ceecdff64/.devcontainer/Dockerfile#L1

Comment (viceice) [reply package.json:133]: i know, it's a todo on pr description 😉

Comment (rarkins) [reply package.json:135]: Still todo, or done?

Comment (viceice) [reply package.json:135]: Still Todo, will cause regular conflicts. will try to do today

Comment (viceice) [reply package.json:135]: done

File: package.json [hunk 2]
```diff
@@ -132,7 +132,7 @@
   },
```
Comment (HonkingGoose) [inline package.json:132]: The `scripts` area in the `package.json` still mentions Node 14:

https://github.com/renovatebot/renovate/blob/0ab15a4b214fb16f69f8c8ece9ae81b438a650f7/package.json#L26

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM containerbase/node:14.20.1@sha256:7772af27efa5703ee593b158e18c4160dff0d7d605d2e48e1d0b5ecdb467556d
+FROM containerbase/node:16.17.1
 
 USER root
```
Comment (HonkingGoose) [inline .devcontainer/Dockerfile:4]: We may also need to edit the GitHub Action files, they still mention Node 14. For example:

https://github.com/renovatebot/renovate/blob/0ab15a4b214fb16f69f8c8ece9ae81b438a650f7/.github/workflows/build.yml#L27

https://github.com/renovatebot/renovate/blob/0ab15a4b214fb16f69f8c8ece9ae81b438a650f7/.github/workflows/build.yml#L41

https://github.com/renovatebot/renovate/blob/0ab15a4b214fb16f69f8c8ece9ae81b438a650f7/.github/workflows/build.yml#L32

Probably a good idea to search for `14` in the `.github` folder, and see what pops up. 😉

**Review by HonkingGoose — COMMENTED** [review]
I found more references to Node 14, maybe those need to be changed/dropped?

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (viceice) [reply .devcontainer/Dockerfile:4]: will be done in a later pr

File: package.json (hunk 2 above)
Comment (viceice) [reply package.json:132]: will cleanup later, as it doesn't change anything

**Pull request merged**